server/src/main/java/org/apache/iotdb/db/auth/StandaloneAuthorityFetcher.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.iotdb.rpc.RpcUtils;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import com.google.common.util.concurrent.SettableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StandaloneAuthorityFetcher implements IAuthorityFetcher {
+
+  private static final Logger logger = LoggerFactory.getLogger(StandaloneAuthorityFetcher.class);
+
+  private LocalConfigNode localConfigNode = LocalConfigNode.getInstance();
+
+  private static final class StandaloneAuthorityFetcherHolder {
+    private static final StandaloneAuthorityFetcher INSTANCE = new StandaloneAuthorityFetcher();
+
+    private StandaloneAuthorityFetcherHolder() {}
+  }
+
+  public static StandaloneAuthorityFetcher getInstance() {
+    return StandaloneAuthorityFetcher.StandaloneAuthorityFetcherHolder.INSTANCE;
+  }
+
+  @Override
+  public TSStatus checkUser(String username, String password) {
+    try {
+      if (localConfigNode.login(username, password)) {
+        return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+      } else {
+        return RpcUtils.getStatus(
+            TSStatusCode.WRONG_LOGIN_PASSWORD_ERROR, "Authentication failed.");
+      }
+    } catch (AuthException e) {
+      return RpcUtils.getStatus(TSStatusCode.AUTHENTICATION_ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  public TSStatus checkUserPrivileges(String username, List<String> allPath, int permission) {
+    boolean checkStatus = true;
+    String checkMessage = null;
+    for (String path : allPath) {
+      try {
+        if (!checkOnePath(username, path, permission)) {
+          checkStatus = false;
+        }
+      } catch (AuthException e) {
+        checkStatus = false;
+      }
+    }
+    if (checkStatus) {
+      return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+    } else {
+      return RpcUtils.getStatus(TSStatusCode.NO_PERMISSION_ERROR, checkMessage);
+    }
+  }
+
+  private boolean checkOnePath(String username, String path, int permission) throws AuthException {
+    try {
+      String fullPath = path == null ? AuthUtils.ROOT_PATH_PRIVILEGE : path;
+      if (localConfigNode.checkUserPrivileges(username, fullPath, permission)) {
+        return true;
+      }
+    } catch (AuthException e) {
+      logger.error("Error occurs when checking the seriesPath {} for user {}", path, username, e);
+      throw new AuthException(e);
+    }
+    return false;
+  }
+
+  @Override
+  public SettableFuture<ConfigTaskResult> operatePermission(
+      TAuthorizerReq authorizerReq, ConfigNodeClient configNodeClient) {
+    SettableFuture<ConfigTaskResult> future = SettableFuture.create();
+    try {
+      LocalConfigNode.getInstance().operatorPermission(authorizerReq);
+    } catch (AuthException e) {
+      future.setException(e);
+    }
+    future.set(new ConfigTaskResult(TSStatusCode.SUCCESS_STATUS));
+    return future;
+  }
```

Comment:
in the final, the future will be set `TSStatusCode.SUCCESS_STATUS` regardless of whether exception happens or not?

The same as others,

Reply:
fixed~